typegen/src/typegen/mod.rs

```diff
@@ -270,4 +271,198 @@ impl<'a> TypeGenerator<'a> {
             derives.insert_derive(parse_quote!(#compact_as_type_path));
         }
     }
+
+    /// Get the type path for a field of a struct or an enum variant, providing any generic
+    /// type parameters from the containing type. This is for identifying where a generic type
+    /// parameter is used in a field type e.g.
+    ///
+    /// ```rust
+    /// struct S<T> {
+    ///     a: T, // `T` is the "parent" type param from the containing type.
+    ///     b: Vec<Option<T>>, // nested use of generic type param `T`.
+    /// }
+    /// ```
+    ///
+    /// This allows generating the correct generic field type paths.
+    ///
+    /// # Panics
+    ///
+    /// If no type with the given id found in the type registry.
```

Comment:
Since we are returning a result here, would it be dificult to make this method not panic? Same for below?

Reply:
Oh good catch! The docs are out of date, it actually returns an Err(..) if a wrong id was provided.